src/ServiceFabric/ServiceFabric/Commands/UpdateAzServiceFabricNodeType.cs

```diff
@@ -0,0 +1,101 @@
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ServiceFabric.Common;
+using Microsoft.Azure.Commands.ServiceFabric.Models;
+using Microsoft.Azure.Management.ServiceFabric;
+using Microsoft.Azure.Management.ServiceFabric.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using ServiceFabricProperties = Microsoft.Azure.Commands.ServiceFabric.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Common.Compute.Version_2018_04;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ServiceFabricNodeType", SupportsShouldProcess = true), OutputType(typeof(PSCluster))]
+    public class UpdateAzServiceFabricNodeType : ServiceFabricNodeTypeCmdletBase
+    {
+        /// <summary>
+        /// Resource group name
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specify the name of the cluster")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("ClusterName")]
+        public override string Name { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true,
+           HelpMessage = "Define whether the node type is a primary node type. Primary node type may have seed nodes and system services.")]
+        public bool? IsPrimaryNodeType { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var cluster = GetCurrentCluster();
+            var existingNodeType = GetNodeType(cluster, this.NodeType, ignoreErrors: true);
+
+            if (existingNodeType == null)
+            {
+                throw new PSArgumentException(
+                    string.Format(
+                        ServiceFabricProperties.Resources.CannotFindTheNodeType,
+                        this.NodeType));
+            }
+
+            if (ShouldProcess(target: this.NodeType, action: string.Format("Update nodetype {0}", this.NodeType)))
+            {
+                if (cluster.NodeTypes == null)
+                {
+                    throw new PSInvalidOperationException(ServiceFabricProperties.Resources.NodeTypesNotDefinedInCluster);
+                }
+
+                if (PatchRequired(existingNodeType))
+                {
+                    if (this.IsPrimaryNodeType.HasValue && this.IsPrimaryNodeType.Value != existingNodeType.IsPrimary)
```

Comment:
do you need to check this again?

Reply:
Good point. I was thinking of this like a staged operation - to check if any change was made at all, then apply changes per parameter (as update node type cmdlet is extended). This could be done another way.